android/src/main/kotlin/io/customer/customer_io/messaginginapp/CustomerIOInAppMessaging.kt

```diff
@@ -0,0 +1,45 @@
+package io.customer.customer_io.messaginginapp
+
+import io.customer.customer_io.CustomerIOPluginModule
+import io.customer.customer_io.constant.Keys
+import io.customer.customer_io.invokeNative
+import io.customer.messaginginapp.di.inAppMessaging
+import io.customer.sdk.CustomerIO
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+
+/**
+ * Flutter module implementation for messaging in-app module in native SDKs. All functionality
+ * linked with the module should be placed here.
+ */
+internal class CustomerIOInAppMessaging(
+    pluginBinding: FlutterPlugin.FlutterPluginBinding,
+) : CustomerIOPluginModule, MethodChannel.MethodCallHandler {
+    override val moduleName: String = "InAppMessaging"
+    private val flutterCommunicationChannel: MethodChannel =
+        MethodChannel(pluginBinding.binaryMessenger, "customer_io_messaging_in_app")
+
+    override fun onAttachedToEngine() {
+        flutterCommunicationChannel.setMethodCallHandler(this)
+    }
+
+    override fun onDetachedFromEngine() {
+        flutterCommunicationChannel.setMethodCallHandler(null)
+    }
```

Comment:
Some of the code in this class are duplicate of what's in `CustomerIOPushMessaging`. 

Could we create an abstract class or something similar to avoid duplication? 

I want to avoid a potential bug in the future where we need to update the `onAttachedToEngine` implementation, for example, and forget to update the implementation in all classes in the SDK.

Reply:
We can cater to this in the later refactor, don't want to make it a part of this release as the refactor would effect `CustomerIOPushMessaging` as well.